Yoast/wordpress-seo
# Remove dependency on 'wp-sanitize' until WP 5.0 is released

## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug where the first tab of the metabox would be empty when using WordPress 4.8.x. 

## Relevant technical choices:

* wp-sanitize is not available on `wp.sanitize` in WordPress 4.8.x. The `stripTags` function we're using in `replacementVariableHelpers.js` (which will be bundled in `wp-seo-post-scraper-xx.min.js`) is part of `press-this.js` in WP 4.8.x.
* Because we only use one small function from `wp-sanitize`, I copied that function to our code base and removed the `wp-sanitize` dependency from the asset manager. As soon as WP 5.0 is released, we can go back to the situation from before this PR.
* Another option is to add logic about which `stripTags` to use depending on which WP version is running. However, because WP 5.0 will be released soon, which means we will drop 4.8.x support, this seemed like a better/more pragmatic solution. 
*I've created an issue to revert this PR after 5.0 has been released: https://github.com/Yoast/wordpress-seo/issues/11732

## Test instructions
This PR can be tested by following these steps:
* When building manually also run `grunt webpack:buildProd` besides regular building steps. 

On trunk:
* Install WP 4.8.x
* Install QueryMonitor
* Edit post. See an empty first tab in the meta box. The same happens for pages, categories etc.
* In the QueryMonitor, you'll see a missing 'wp-sanitize' dependency needed for the post-scraper/term-scraper.

On this branch:
* On WP 4.8.x: 
     * Edit post/page/category. See a fully functional meta box. 
     * Use the `exerpt` replacevar but don't write an actual excerpt in the excerpt input field. Write some content containing HTML tags. 
     * Both the snippet editor and the frontend should show a metadescription containing the first 156 characters of your text  without HTML tags (or a little less, depending on word boundaries).
* On WP 4.9.8 and WP 5.0: 
     * Edit post/page/category. See a fully functional meta box. 
     * Use the `exerpt` replacevar but don't write an actual excerpt in the excerpt input field. Write some content containing HTML tags. 
     * Both the snippet editor and the frontend should show a metadescription containing the first 156 characters of your text without HTML tags (or a little less, depending on word boundaries).

## Quality assurance


Fixes #11706
